### PR TITLE
feat(useUrlSearchParams): add `write` option

### PR DESCRIPTION
**Pull request by waldonUB**
### Additional context

fix [#2164](https://github.com/vueuse/vueuse/issues/2164)

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useUrlSearchParams/index.ts [hunk 1]
```diff
@@ -21,6 +21,11 @@ export interface UseUrlSearchParamsOptions<T> extends ConfigurableWindow {
    * @default {}
    */
   initialValue?: T
+
+  /**
+   * @default false
+   */
+   onlyRead?: boolean
```
Comment (antfu) [inline packages/core/useUrlSearchParams/index.ts:28]:
```suggestion
  /**
   * Write back to `window.history` automatically
   *
   * @default true
   */
  write?: boolean
```

Comment (waldonUB) [reply packages/core/useUrlSearchParams/index.ts]: Thanks for review. I have revised it as suggested. The variable is called `write` has been declared below so I changed it to `autoWrite`

File: packages/core/useUrlSearchParams/index.ts [hunk 2]
```diff
@@ -21,6 +21,13 @@ export interface UseUrlSearchParamsOptions<T> extends ConfigurableWindow {
    * @default {}
    */
   initialValue?: T
+
+  /**
+   * Write back to `window.history` automatically
+   *
+   * @default true
+   */
+   autoWrite?: boolean
```
Comment (chris-zhu) [inline packages/core/useUrlSearchParams/index.ts:30]: Remove extra spaces for alignment.

**Review by chris-zhu — CHANGES_REQUESTED** [review]

File: packages/core/useUrlSearchParams/index.ts (hunk 2 above)
Comment (chris-zhu) [inline packages/core/useUrlSearchParams/index.ts:30]:
```suggestion
 autoWrite?: boolean
```

**Review by chris-zhu — CHANGES_REQUESTED** [review]

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
Please fix lint.

**Review by antfu — APPROVED** [review]

**Pull request merged**